### ISSUE TITLE
Revisit duplication of photon model code

### ISSUE THREAD

**Issue by jbphet** [phetsims/greenhouse-effect#111]
For issue https://github.com/phetsims/greenhouse-effect/issues/108 I removed a mixin that was being used to add common photon modeling code to the models for the "Photons" and "Layer Model" screens.  I did this largely because the mixin wasn't working very well, PhET hasn't completely settled on a standard pattern for mixins in TypeScript, and because my experience with mixins is that they tend to confuse the IDE, so it's good to avoid them if possible.

I just realized that there was a bit of a negative consequence to this refactoring, which was that the `LayerModelModel` class had `ConcentrationModel` in its lineage, which wasn't appropriate because it isn't using layers to model the concentration of greenhouse gasses in the atmosphere, it's instead showing the layers explicitly.  In 1facf09774da2fa93a227a9c0dd2766591024551, I pulled the photon modeling code into the two classes that needed it and got rid of the class where it had been centralized.  This allowed the `LayerModelModel` class to extend a base class higher up in the hierarchy that doesn't have the unnecessary gas-concentration-related code, but it means some of the modeling code is now duplicated.  This may be a good thing long term, because there is some discussion on the design team that the photon behavior may end up being a little different in these two screens.  However, we don't know yet, and are in the process of iterating on this behavior.

If, in the end, the code ends up being the essentially the same in `LayerModelModel` and `PhotonsModel`, it should be pulled back into a mixin or some sort of class that could be reused via composition so that there isn't any code duplication.

I'll assign this to myself, defer, and put a reminder on my calendar to check the situation is a couple of months.

**Comment by jbphet** [comment]
Undeferring.  I'll take a look at this prior to the next prototype publication.

**Comment by jbphet** [comment]
I just took at the code and inheritance hierarchies for the `PhotonsModel` and `LayerModelModel` classes, and I think the way this ended up is pretty decent.  Both of the classes are less than 200 lines long total, and they get their layer modeling functionality from the `LayerModel` class.  The hierarchies look like this:

![image](https://github.com/phetsims/greenhouse-effect/assets/3082831/7beae361-a751-46ce-a448-b0346010524a)

This organizational structure seems reasonable and the layer modeling code is centralized in the `LayersModel` base class.  I improved the documentation a bit, and I feel like that should do it.  Closing.
